### PR TITLE
[server] Remove dead permissions code

### PR DESCRIPTION
**Pull request by geropl**
## Description

Cleanup permissions while going through the existing code to prepare a suggestion for FGA.

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-perms-cleanup</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-perms-cleanup.preview.gitpod-dev.com/workspaces" target="_blank">gpl-perms-cleanup.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/gitpod-protocol/src/permission.ts [hunk 1]
```diff
@@ -6,18 +6,13 @@
 
 // see below for explanation
 export const Permissions = {
-    monitor: undefined,
     developer: undefined,
-    enforcement: undefined,
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:11]: This was used to guard the `/api/enforcement/*` endpoints, which no longer exist.

File: components/gitpod-protocol/src/permission.ts [hunk 2]
```diff
@@ -6,18 +6,13 @@
...
-    "privileged-ws": undefined,
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:12]: The usage of this is gone for years :upside_down_face:

File: components/gitpod-protocol/src/permission.ts [hunk 3]
```diff
@@ -6,18 +6,13 @@
 
 // see below for explanation
 export const Permissions = {
-    monitor: undefined,
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:9]: This was used to grant access to monitor.gitpod.io, which does not exist anymore for years already.

File: components/gitpod-protocol/src/permission.ts [hunk 4]
```diff
@@ -6,18 +6,13 @@
...
     developer: undefined,
-    enforcement: undefined,
-    "privileged-ws": undefined,
     "registry-access": undefined,
     "admin-permissions": undefined,
     "admin-users": undefined,
     "admin-workspace-content": undefined,
     "admin-workspaces": undefined,
     "admin-projects": undefined,
-    "admin-api": undefined,
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:19]: This is got superseded by the more fine-grained roles above :point_up:

File: components/gitpod-protocol/src/permission.ts [hunk 5]
```diff
@@ -6,18 +6,13 @@
...
-    "ide-settings": undefined,
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:20]: Not sure how this was used.... @akosyakov You probably no? Still relevant somehow? :thinking:

Comment (akosyakov) [reply components/gitpod-protocol/src/permission.ts:20]: I think it was a feature flag to allow custom IDE docker images 2-3 years ago. It should not be relevant anymore.

File: components/gitpod-protocol/src/permission.ts [hunk 6]
```diff
@@ -53,15 +48,9 @@ export namespace RolesOrPermissions {
...
-    export const MONITOR: PermissionName = "monitor";
-
     /** The permission to develop on this running Gitpod installation */
     export const DEVELOPER: PermissionName = "developer";
 
-    /** The permission for actions like block user, stop workspace, etc. */
-    export const ENFORCEMENT: PermissionName = "enforcement";
-
     /** The permission for registry access (start workspaces referencing gitpod-internal Docker images) */
     export const REGISTRY_ACCESS: PermissionName = "registry-access";
```
Comment (geropl) [inline components/gitpod-protocol/src/permission.ts:55]: Is this still used/required? @WVerlaek Maybe you can tell? Only usage seems to be [here](https://github.com/gitpod-io/gitpod/blob/097d178910f860d932d3c5343a334db156e7d948/components/server/src/workspace/workspace-starter.ts#L1003). :thinking:

Comment (WVerlaek) [reply components/gitpod-protocol/src/permission.ts:55]: Not sure, but seems to be used for the DevOps role [here](https://github.com/gitpod-io/gitpod/blob/1a663acfd31c1c8c7df0a25cebc1abfd66cbcce7/components/gitpod-protocol/src/permission.ts#L102)? I don't know if that's in use though

Comment (geropl) [reply components/gitpod-protocol/src/permission.ts:55]: I checked the DB, and the occurrences there seemed to be leftovers only.

I was more interested of the impact/usage of the [BuildRegistryAuthTotal](https://github.com/gitpod-io/gitpod/blob/097d178910f860d932d3c5343a334db156e7d948/components/server/src/workspace/workspace-starter.ts#L1005-L1006C62)v construct: Is that implemented in image-builder-mk3 at all? Is it used anywhere concretely?

**Review by easyCZ — APPROVED** [review]

**Review by WVerlaek — APPROVED** [review]

File: components/gitpod-protocol/src/permission.ts (hunk 6 above)
Comment (geropl) [reply components/gitpod-protocol/src/permission.ts:55]: > do you know if that role is also no longer in use

It still has some entries in the prod DB, but they look like leftovers. But it's a role, only an aggregation of permissions, so we can replace it with "registry-access" if we really need to.
We might have to ask Chris on why we have it in the first place... :+1:

**Pull request merged**